sentinelhub/data_utils.py

```diff
@@ -161,11 +156,12 @@ def get_failed_statistical_requests(result_data: List[JsonDict]) -> List[JsonDic
     :return: Failed requests of (Batch) Statistical Results.
     """
     if _is_batch_stat(result_data[0]):
+        batch_failed_response = ((result["identifier"], _get_failed_batch_response(result)) for result in result_data)
         return [
-            {"identifier": result["identifier"], "failed_intervals": _get_failed_intervals(result)}
-            for result in result_data
-            if _get_failed_intervals(result)
+            {"identifier": ident, "failed_intervals": intervals}
+            for ident, intervals in batch_failed_response
+            if intervals != []
         ]
 
-    failed_intervals = _get_failed_intervals(result_data[0])
-    return [{"failed_intervals": failed_intervals}] if failed_intervals else []
+    failed_intervals = _FULL_TIME_RANGE if "error" in result_data[0] else _get_failed_intervals(result_data[0]["data"])
```

Comment:
why do we always look at the zero element here? why are we sure there is no others?

Reply:
If it is a normal statistical request there will be only one result in the list

Reply:
but it would be almost no extra work to make it work on any list (and we then don't need to worry about the implicit assumption that there will always be just 1)

Reply:
I make it go through the list and use a defult index as the identifier for normal stat